src/Source/CsvSourceIterator.php

```diff
@@ -145,13 +145,77 @@ public function rewind()
     public function valid()
     {
         if (!is_array($this->currentLine)) {
-            if (is_resource($this->file)) {
-                fclose($this->file);
-            }
-
             return false;
         }
 
         return true;
     }
+
+    /**
+     * Initialize read process setting CSV options
+     * and settings field names. (Assuming that first line is column name).
+     */
+    private function initializeRead()
+    {
+        // close the previous file
+        if ($this->file instanceof \SplFileObject) {
+            $this->file = null;
+        }
```

Comment:
If this closes the previous file just by loosing the last reference to it, how does line 165 not have the exact same effect?